File: src/main/java/com/Khata/Khata/Service/GigService.java
package com.Khata.Khata.Service;

import com.Khata.Khata.Entity.Gig;
import com.Khata.Khata.Repository.GigRepository;
import org.springframework.stereotype.Service;

@Service
public class GigService {

    private GigRepository gigRepository;

    GigService(GigRepository gigRepository)
    {
        this.gigRepository = gigRepository;
    }

    public Gig addGig(Gig gig)
    {
        Gig savedGig = gigRepository.save(gig);
        return savedGig;
    }
}

File: classtracker/src/main/kotlin/com/classtrackeer/classtracker/Service/GigService.kt
package com.classtrackeer.classtracker.Service

import com.classtrackeer.classtracker.Model.Gig
import com.classtrackeer.classtracker.Repository.GigRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service

@Service
class GigService (@Autowired val gigRepository : GigRepository){

    fun addGig(gig : Gig) : Gig {
        return gigRepository.save(gig)
    }
}